zebra-network/src/meta_addr.rs

```diff
@@ -866,6 +873,10 @@ impl Ord for MetaAddr {
         // First, try states that are more likely to work
         let more_reliable_state = self.last_connection_state.cmp(&other.last_connection_state);
 
+        // Then, try addresses that are more likely to be valid.
+        // Currently, this prefers addresses with canonical Zcash ports.
+        let more_likely_valid = self.peer_preference().cmp(&other.peer_preference());
```

Comment:
Prefer address book peers on canonical addresses.

(Ok sorts before Err, and Preferred sorts before Acceptable.)

Reply:
Please check https://github.com/ZcashFoundation/zebra/pull/4564#discussion_r890471136.